### ISSUE TITLE
Field initialization and @:autoBuild

### ISSUE THREAD

**Issue by mcheshkov** [HaxeFoundation/haxe#4790]
I'm sorry for quite long example.

``` haxe
package builder;

#if macro
import haxe.macro.Context;
import haxe.macro.Expr;
import haxe.macro.Type;
#end

#if !macro
@:autoBuild(builder.Builder.build())
#end
interface IBuilder {}

class Builder {
    macro static public function build():Array<Field> {
        var cCur = Context.getLocalClass().get();

        function loop(c:ClassType) {
            var fs = c.fields.get();
            if (c.superClass != null) loop(c.superClass.t.get());
        }
        loop(cCur);

        return null;
    }
}

```

``` haxe
package consts;

class Consts {
    public static var str:String = "TEST";
}
```

``` haxe
package user1;

import consts.Consts;
import builder.Builder;

class User1 implements IBuilder {
    var STR:String = Consts.str;
    var STR2:String;
    function f(){
            STR2 = Consts.str;
            var STR3 = Consts.str;
    }
}
```

``` haxe
package user2;

import user1.User1;

class User2 extends User1 {
}
```

``` haxe
package ;

import user2.User2;

class Main {
    public static function main(){}
}

```

This give me `user1/User1.hx:7: characters 18-28 : Class<consts.Consts> has no field str`.
Remove `var STR:String = Consts.str;` line - works good.
Remove `c.fields.get` call - works good.
Import `User1` instead of `User2` in `Main` - works good.
That's Haxe 3.2.1 on Linux.

**Comment by Simn** [comment]
Could you try this again on current git version?

**Comment by Simn** [comment]
I've checked and this still fails. There seems to be a clear build order here:

1. Main
2. User2
3. User1
4. Consts

The fact that removing the `var STR:String = Consts.str;` makes the error go away suggests that it's a problem with the pass flushing.

I don't know what to do with this right now, it might not be possible to address this for 3.3.

**Comment by RealyUniqueName** [comment]
Fixed with #8410
